apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -45,71 +26,41 @@ require('/views/shared/test/unit/mock_utils.js');
 require('/views/shared/js/activity_handler.js');
 
 var mocksHelperForActivityHandler = new MocksHelper([
+  'ActivityClient',
+  'ActivityShim',
   'Attachment',
-  'Contacts',
   'Draft',
   'Drafts',
   'MessageManager',
-  'NotificationHelper',
-  'Notify',
-  'Settings',
-  'SettingsURL',
-  'SilentSms',
-  'SMIL',
-  'Threads',
-  'Utils',
   'Navigation',
-  'ActivityShim',
-  'ActivityClient'
+  'Settings',
+  'Utils'
 ]).init();
 
 suite('ActivityHandler', function() {
   mocksHelperForActivityHandler.attachTestHelpers();
 
-  var realSetMessageHandler;
-  var realWakeLock;
-  var realMozApps;
-  var realMozL10n;
-
-  var isDocumentHidden;
+  var navigatorMocks = new Map([
```

Comment:
It's my new experiment with navigator mocks, I always didn't like old approach, so trying to make it easier (it's better visible in system_message_handler_tests.js - we have several navigator mocks there, here I use Map for one mock only, just to be consistent with system_message_handler_tests). It was actually quite cumbersome to make old mocks work with Promises (e.g. mozApps mock), that is one of the reasons I've tried to use new approach :)

Would be happy to hear your thoughts or other ideas!